src/core/model_concrete.jl

```diff
@@ -0,0 +1,257 @@
+
+A concrete version of [`AgentBasedModel`](@ref) that stores all agents in a
+single container. Offers the variants:
+
+- [`StandardABM`](@ref)
+- [`UnkillableABM`](@ref)
+- [`FixedMassABM`](@ref)
+"""
+function SingleContainerABM(
+    ::Type{A},
+    space::S = nothing;
+    container::Type = Dict{Int,A},
+    scheduler::F = Schedulers.fastest,
+    properties::P = nothing,
+    rng::R = Random.default_rng(),
+    warn = true
+) where {A<:AbstractAgent,S<:SpaceType,F,P,R<:AbstractRNG}
+    agent_validator(A, space, warn)
+    C = construct_agent_container(container, A)
+    agents = C()
+    return SingleContainerABM{S,A,C,F,P,R}(agents, space, scheduler, properties, rng, Ref(0))
+end
+
+function SingleContainerABM(agent::AbstractAgent, args...; kwargs...)
+    return SingleContainerABM(typeof(agent), args...; kwargs...)
+end
+
+
+"""
+    StandardABM(AgentType [, space]; properties, kwargs...) → model
+
+
+The most standard concrete implementation of an [`AgentBasedModel`](@ref),
+as well as the default version of the generic [`AgentBasedModel`](@ref) constructor.
+`StandardABM` stores agents in a dictionary mapping unique `Int` IDs to agents.
+See also [`UnkillableABM`](@ref), [`FixedMassABM`](@ref).
+"""
+StandardABM(args...; kwargs...) = SingleContainerABM(args...; kwargs..., container=Dict{Int})
+
+"""
+    UnkillableABM(AgentType [, space]; properties, kwargs...) → model
+
+Similar to [`StandardABM`](@ref), but agents cannot be removed, only added.
+This allows storing agents more efficiently in a standard Julia `Vector` (as opposed to
+the `Dict` used by [`StandardABM`](@ref), yielding faster retrieval and iteration over agents.
+
+It is mandatory that the agent ID is exactly the same as the agent insertion
+order (i.e., the 5th agent added to the model must have ID 5). If not,
+an error will be thrown by [`add_agent!`](@ref).
+"""
+UnkillableABM(args...; kwargs...) = SingleContainerABM(args...; kwargs..., container=Vector)
+
+"""
+    FixedMassABM(agent_vector [, space]; properties, kwargs...) → model
+
+Similar to [`UnkillableABM`](@ref), but agents cannot be removed nor added.
+Hence, all agents in the model must be provided in advance as a vector.
+This allows storing agents into a `SizedVector`, a special vector with statically typed
+size which is the same as the size of the input `agent_vector`.
+This version of agent based model offers better performance than [`UnkillableABM`](@ref)
+if the number of agents is important and used often in the simulation.
+
+It is mandatory that the agent ID is exactly the same as its position
+in the given `agent_vector`.
+"""
+function FixedMassABM(
+    agents::AbstractVector{A},
+    space::S = nothing;
+    scheduler::F = Schedulers.fastest,
+    properties::P = nothing,
+    rng::R = Random.default_rng(),
+    warn = true
+) where {A<:AbstractAgent, S<:SpaceType,F,P,R<:AbstractRNG}
+    C = SizedVector{length(agents), A}
```

Comment:
If we have `construct_agent_container`, shouldn't we aim to use it consistently?

```julia
construct_agent_container(T::Type{<:SizedVector}, A) = T{A}
```

Then call it as `construct_agent_container(SizedVector{length(agents)}, A)`

Reply:
The problem is that the Sized case is rather special and initializes a container that already has all agents instead of an emptu container